src/parse/asp/interpreter.go

```diff
@@ -34,10 +34,15 @@ func newInterpreter(state *core.BuildState, p *Parser) *interpreter {
 		state:  state,
 		locals: map[string]pyObject{},
 	}
+	// If we're creating an interpreter for a subrepo, we should share the subinclude cache.
+	subincludes := &subincludeMap{m: map[string]subincludeResult{}}
+	if p.interpreter != nil {
+		subincludes = p.interpreter.subincludes
+	}
```

Comment:
This also doesn't fix any race conditions. It just means we don't duplicate work between subrepos.